app/main.py

```diff
@@ -1,3 +1,80 @@
+from typing import Any
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: int) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Any) -> "Distance":
+        if isinstance(other, Distance):
+            return Distance(self.km + other.km)
+        if isinstance(other, (int, float)):
+            return Distance(self.km + other)
+        raise TypeError("Addition only supported for\n"
+                        "Distance or numeric types.")
+
+    def __iadd__(self, other: Any) -> "Distance":
+        if isinstance(other, Distance):
+            self.km += other.km
+        elif isinstance(other, (int, float)):
+            self.km += other
+        else:
+            raise TypeError("Addition only supported for\n"
+                            "Distance or numeric types.")
+        return self
+
+    def __mul__(self, factor: Any) -> "Distance":
+        if isinstance(factor, (int, float)):
+            return Distance(self.km * factor)
+        raise TypeError("Multiplication only supported with numeric types.")
+
+    def __truediv__(self, divisor: Any) -> "Distance":
+        if isinstance(divisor, (int, float)) and divisor != 0:
+            return Distance(round(self.km // divisor, 2))
```

Comment:
The `__truediv__` method uses integer division (`//`) which is incorrect for true division. It should use `/` for floating-point division to ensure the result is a float. Additionally, rounding should be applied to the result of the division, not to the integer division.